src/lib/form/validation.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * @typedef {import('./validation.types.js').ValidValidity} ValidValidity
+ * @typedef {import('./validation.types.js').InvalidValidity} InvalidValidity
+ * @typedef {import('./validation.types.js').Validator} Validator
+ * @typedef {import('./validation.types.js').Validity} Validity
+ * @typedef {import('./validation.types.js').ErrorMessage} ErrorMessage
+ */
+
+export class Validation {
+  /** @type {ValidValidity} */
+  static VALID = { valid: true };
+
+  /**
+   * @param {string} code
+   * @return {InvalidValidity}
+   */
+  static invalid (code) {
+    return {
+      valid: false,
+      code,
+    };
+  }
+}
+
+/**
+ * A validator that checks whether a value is empty or not.
+ *
+ * It considers an empty value as invalid with `empty` error code.
+ *
+ * It supports emptiness for string, number and array data types.
+ *
+ * @implements {Validator}
+ */
+export class RequiredValidator {
+  /**
+   * @param {any} value
+   * @param {Object} _formData
+   * @return {Validity}
+   */
+  validate (value, _formData) {
```

Comment:
This is because we told that the RequiredValidator implements a Validator. When implementing a method, the signature must be respected. Webstorm was complaining about that.

But you're right, typescript is ok with that so I could remove the parameter